src/rail_gui.cpp

```diff
@@ -1480,6 +1482,7 @@ struct BuildSignalWindow : public PickerWindowBase {
 private:
 	Dimension sig_sprite_size;     ///< Maximum size of signal GUI sprites.
 	int sig_sprite_bottom_offset;  ///< Maximum extent of signal GUI sprite from reference point towards bottom.
+	bool show_non_path_signals;    ///< Whether non-path signals are shown
```

Comment:
This is only used in SetSignalUIMode. No need for it to be a member variable at all?